src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -118,4 +114,47 @@ void AsyncOpenTask::unregister_download_progress_notifier(uint64_t token)
         m_session->unregister_progress_notifier(token);
 }
 
+void AsyncOpenTask::attach_to_subscription_initializer(AsyncOpenCallback&& async_open_callback,
+                                                       std::shared_ptr<_impl::RealmCoordinator> coordinator)
+{
+    // Subscription initialization will occure in either of these 2 cases.
+    // 1. The realm file has been created (this is the first time we ever open the file)
+    // 2. we are instructed to run the subscription initializer via sync_config->rerun_init_subscription_on_open.
+    //    But we do that only if this is the first time we open realm.
+
+    // subscription initializer called here. We need to wait until it completes
+    auto shared_realm = coordinator->get_realm();
+    std::shared_ptr<AsyncOpenTask> self(shared_from_this());
+    const auto init_subscription = shared_realm->get_latest_subscription_set();
+    init_subscription.get_state_change_notification(sync::SubscriptionSet::State::Complete)
```

Comment:
We need a check here unfortunately.

Consider the following case:
1.  Async open + initial subscription. Wait to complete
2. User goes offline. Commits other subscriptions
3. Other users commit data which is in view for the initial subscription
4. User comes back online
5. User should download latest data available, but it now waits until the offline subscriptions (and any other local changes) are uploaded and data is bootstrapped